ios/FluentUI/Core/SwiftUI+ViewModifiers.swift

```diff
@@ -5,6 +5,38 @@
 
 import SwiftUI
 
+struct SquareShapedViewModifier: ViewModifier {
+    func body(content: Content) -> some View {
+        let modifiedContent = HStack {
+            content
+                .alignmentGuide(HorizontalAlignment.center) { (viewDimensions) -> CGFloat in
+                    DispatchQueue.main.async {
+                        self.size = max(size, // ensures the View does not shrink
+                                        min(maxSize,
+                                            max(minSize,
+                                                max(viewDimensions.height,
+                                                    viewDimensions.width))))
```

Comment:
I'm not sure if it's just me, but this chain of max/min/max/max is a bit confusing. I understand what's going on after thinking about it for a bit, but it might help if we had a comment with a longer explanation.